rustls/src/server/tls13.rs

```diff
@@ -1157,13 +1157,15 @@ impl<C: CryptoProvider> State<ServerConnectionData> for ExpectFinished<C> {
             .key_schedule
             .sign_client_finish(&handshake_hash, cx.common);
 
-        let fin = constant_time::verify_slices_are_equal(expect_verify_data.as_ref(), &finished.0)
-            .map_err(|_| {
+        let fin = match ConstantTimeEq::ct_eq(expect_verify_data.as_ref(), &finished.0[..]).into() {
+            true => verify::FinishedMessageVerified::assertion(),
+            false => {
                 warn!("Finished wrong");
```

Comment:
I think I'm minded to remove this logging. Like other uses of `DecryptError` it points to data corruption (malicious or otherwise) between us and the peer. That's not actionable for us or the peer, so I'm not sure the logging is useful?

Reply:
That makes sense to me :+1: